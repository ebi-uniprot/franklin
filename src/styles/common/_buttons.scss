@import '../settings';
@import '../colours';

.button {
  svg {
    width: 0.9rem;
    height: 0.9rem;
    margin-top: -0.2rem;
    margin-right: 0.5rem;
    vertical-align: middle;
    fill: $colour-sky-white;
  }
  &.large-icon svg {
    width: 1.2rem;
    height: 1.2rem;
    margin-top: -0.2rem;
    margin-right: 0.5rem;
  }
  &.secondary {
    color: $colour-sea-blue;
    border: 0.1rem solid $colour-sea-blue;
    font-weight: 700;
    svg {
      fill: $colour-sea-blue;
    }
  }
  &.tertiary {
    background: none;
    color: $colour-sapphire-blue;
    font-weight: $global-weight-bold;
    padding: 0;
    line-height: 1.6;
    text-align: left;
    &:hover:disabled {
      color: $colour-sapphire-blue;
      background-color: $colour-sky-white;
    }
    svg {
      fill: $colour-sapphire-blue;
    }
    &.tertiary-icon__active {
      svg {
        fill: $colour-platinum;
      }
    }
  }
}

.button-group {
  margin-bottom: $global-margin/2;
  .tertiary {
    margin-right: 1rem;
  }
}
