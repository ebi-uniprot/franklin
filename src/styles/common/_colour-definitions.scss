@import '../colours';

// PRIMARY
.colour-primary-500 {
  background: map-get($franklin-colours, colourPrimary500);
  color: map-get($franklin-colours, colourGreyscale100);
}
.colour-primary-400 {
  background: map-get($franklin-colours, colourPrimary400);
  color: map-get($franklin-colours, colourGreyscale100);
}
.colour-primary-300 {
  background: map-get($franklin-colours, colourPrimary300);
  color: map-get($franklin-colours, colourGreyscale100);
}
.colour-primary-200 {
  background: map-get($franklin-colours, colourPrimary200);
  color: map-get($franklin-colours, colourGreyscale600);
}
.colour-primary-100 {
  background: map-get($franklin-colours, colourPrimary100);
  color: map-get($franklin-colours, colourGreyscale600);
}

// GREY
.colour-greyscale-600 {
  background: map-get($franklin-colours, colourGreyscale600);
  color: map-get($franklin-colours, colourGreyscale100);
}
.colour-greyscale-500 {
  background: map-get($franklin-colours, colourGreyscale500);
  color: map-get($franklin-colours, colourGreyscale100);
}
.colour-greyscale-400 {
  background: map-get($franklin-colours, colourGreyscale400);
  color: map-get($franklin-colours, colourGreyscale100);
}
.colour-greyscale-300 {
  background: map-get($franklin-colours, colourGreyscale300);
  color: map-get($franklin-colours, colourGreyscale600);
}
.colour-greyscale-200 {
  background: map-get($franklin-colours, colourGreyscale200);
  color: map-get($franklin-colours, colourGreyscale600);
}
.colour-greyscale-100 {
  background: map-get($franklin-colours, colourGreyscale100);
  color: map-get($franklin-colours, colourGreyscale600);
}
// CURATION
.colour-reviewed {
  background: map-get($franklin-colours, colourReviewed);
  color: map-get($franklin-colours, colourGreyscale600);
}
.colour-unreviewed {
  background: map-get($franklin-colours, colourUnreviewed);
  color: map-get($franklin-colours, colourGreyscale600);
}
// NAMESPACE
.colour-uniprotkb {
  background: map-get($franklin-colours, colourUniprotkb);
  color: map-get($franklin-colours, colourGreyscale600);
}
.colour-uniref {
  background: map-get($franklin-colours, colourUniref);
  color: map-get($franklin-colours, colourGreyscale600);
}
.colour-uniparc {
  background: map-get($franklin-colours, colourUniparc);
  color: map-get($franklin-colours, colourGreyscale600);
}
.colour-proteomes {
  background: map-get($franklin-colours, colourProteomes);
  color: map-get($franklin-colours, colourGreyscale100);
}
.colour-supporting-data {
  background: map-get($franklin-colours, colourSupportingData);
  color: map-get($franklin-colours, colourGreyscale100);
}
.colour-peptide-search {
  background: map-get($franklin-colours, colourPeptideSearch);
  color: map-get($franklin-colours, colourGreyscale100);
}
.colour-id-mapping {
  background: map-get($franklin-colours, colourIdMapping);
  color: map-get($franklin-colours, colourGreyscale100);
}
.colour-blast {
  background: map-get($franklin-colours, colourBlast);
  color: map-get($franklin-colours, colourGreyscale100);
}
.colour-align {
  background: map-get($franklin-colours, colourAlign);
  color: map-get($franklin-colours, colourGreyscale100);
}

// svg fills
.svg-colour-reviewed svg {
  color: map-get($franklin-colours, colourReviewed);
}
.svg-colour-unreviewed svg {
  color: map-get($franklin-colours, colourUnreviewed);
}

// messages
.colour-warning {
  background: map-get($franklin-colours, colourWarning);
}
.colour-failure {
  background: map-get($franklin-colours, colourFailure);
}
.colour-success {
  background: map-get($franklin-colours, colourSuccess);
}
.colour-info {
  background: map-get($franklin-colours, colourInfo);
}

// Data visualisation
.colour-coyote-brown {
  background: map-get($franklin-colours, colourDataVis1);
}
.colour-outer-space {
  background: map-get($franklin-colours, colourDataVis2);
}

.box-colour {
  vertical-align: top;
  display: inline-block;
  width: 9em;
  height: 9em;
  margin-bottom: 4em;
  font-size: $small-font-size;
  &:before {
    content: attr(data-name);
    color: map-get($franklin-colours, colourGreyscale600);
    margin-top: 10em;
    display: block;
    text-align: center;
  }
}
