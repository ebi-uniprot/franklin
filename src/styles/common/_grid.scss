@import '../settings';
@import '../colours.js';

$grid-name: 'uniprot-grid';
$gap: 1rem;
$max-width: 117ch; // 1070px with Lato 16px base as per mockup
$n-columns: 12;

/* uniprot-website grid system */
.#{$grid-name} {
  display: grid;
  grid-column-gap: $gap; /* or we can implement that differently */
  grid-template-columns: repeat(12, 1fr);

  & > .#{$grid-name}-cell {
    @for $i from 1 through $n-columns {
      &--span-#{$i} {
        grid-column: auto / span $i;
      }
    }

    @for $i from 1 through $n-columns {
      &--offset-#{$i} {
        grid-column-start: $i + 1;
      }
    }
  }
}

.#{$grid-name}--centered {
  margin-left: auto;
  width: calc(min(100vw, #{$max-width}));
  margin-right: auto;
}

.#{$grid-name}--with-bleed {
  box-shadow: -$gap 0 var(--grid-background), $gap 0 var(--grid-background);
}
