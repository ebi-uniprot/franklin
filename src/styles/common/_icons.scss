@import '../settings';
@import '../colours';

.icon {
  &--reviewed {
    color: $colour-reviewed;
  }
  &--unreviewed {
    color: $colour-unreviewed;
  }
}
