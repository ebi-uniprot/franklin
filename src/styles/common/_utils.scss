@import '../settings';

$small-bubble: 1.6rem;
$medium-bubble: 3rem;
$large-bubble: 5rem;

$rootPx: 16px;

@function remToPx($rem) {
  @return unitless-calc($rem) * $rootPx;
}

@function pxToRem($px) {
  @return (unitless-calc($px) / unitless-calc($rootPx)) + 0px;
}

@function bubbleFontSize($size) {
  @return $size/4 + 1/3;
}
