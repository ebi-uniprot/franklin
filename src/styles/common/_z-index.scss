$z-index-low: 2000;
$z-index-medium: 5000;
$z-index-high: 9000;
$z-index-highest: 10000;
