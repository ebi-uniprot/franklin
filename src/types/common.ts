import { CSSProperties } from 'react';

export type HeadingLevels = `h${1 | 2 | 3 | 4 | 5 | 6}`;
export interface FranklinStyle extends CSSProperties {
  '--main-button-color': string;
}
