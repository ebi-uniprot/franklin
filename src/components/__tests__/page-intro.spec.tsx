import { render, screen } from '@testing-library/react';
import { HeadingLevels } from '../../types/common';

import PageIntro from '../page-intro';

describe('PageIntro component', () => {
  test('should render', () => {
    const { asFragment } = render(
      <PageIntro title="Title" resultsCount={1000}>
        <div>Some content</div>
      </PageIntro>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test('should render with heading level', () => {
    const level = 5;
    const headingLevel = `h${level}` as HeadingLevels;
    render(
      <PageIntro title="Title" resultsCount={1000} headingLevel={headingLevel}>
        <div>Some content</div>
      </PageIntro>
    );
    expect(screen.getByRole('heading', { level })).toBeInTheDocument();
  });

  test('should render title postscript', () => {
    render(
      <PageIntro
        title="Title"
        resultsCount={1000}
        titlePostscript={<i>from job ID123</i>}
      />
    );
    expect(
      screen.getByRole('heading', {
        name: /Title 1,000 results from job ID123/i,
      })
    ).toBeInTheDocument();
  });
});
