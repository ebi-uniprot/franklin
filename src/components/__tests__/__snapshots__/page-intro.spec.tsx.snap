// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageIntro component should render 1`] = `
<DocumentFragment>
  <div
    class="page-intro"
  >
    <h2>
      <button
        class="dropdown"
        type="button"
      >
        <test-file-stub
          height="20"
          width="20"
        />
         Title
      </button>
      <small>
         1,000 results 
      </small>
    </h2>
    <div
      class="intro-content"
    >
      <div>
        Some content
      </div>
      <div
        class="intro-links"
      />
    </div>
  </div>
</DocumentFragment>
`;
